front/components/assistant_builder/InstructionScreen.tsx

```diff
@@ -251,6 +280,26 @@ export function InstructionScreen({
       </div>
       <div className="flex h-full flex-col gap-1">
         <div className="relative h-full min-h-[240px] grow gap-1 p-px">
+          <div className="absolute right-2 top-2 z-10">
+            {agentConfigurationHistory &&
+              agentConfigurationHistory.length > 1 && (
+                <PromptHistory
+                  history={configsWithUniqueInstructions}
+                  onVersionPick={(config, isLatest: boolean) => {
+                    // Only allow edition of latest version, because no memory or other versions.
```

Comment:
Not sure to understand this, very confusing for the user.
Do we maybe want something more of a dialog that says 'reset instructions to version from {date}?' and that lets us keep editing?

cc @pinotalexandre

Reply:
Agree that this is not ideal here is the dilemma:
- We want users to be able to jump to previous versions, copy parts, and paste back to their current draft. We expect this to be more likely than to start again from an older version entirely.
- We need to stick to a wysiwyg editor -> broken in current version and confusing (as you mentioned) because you can't edit and what is saved is not necessary the version you're currently inspecting

Reply:
If we decide not to just overwrite the current instructions and let the user keep editing, then maybe we need to show the previous version in some sort of modal so they can copy/paste stuff

Because for now I just feel like I lose my current work and I can't keep editing because the textarea is disabled, so I just don't know what to do :')

Reply:
What if we actually remember what the user typed so that you can come back to your work on a different version (saved) any time ? Should solve both problems at reasonable cost